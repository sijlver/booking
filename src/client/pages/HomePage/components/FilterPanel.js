import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Field } from '../../../components';
import helperFunction from '../../../utils';

class FilterPanel extends Component {
    state = {
        searchName: '',
        dateStart: '',
        dateEnd: '',
        limit: '',
        minStartDate: helperFunction.changeFormatDate(new Date()),
        minEndDate: helperFunction.changeFormatDate(new Date(), 1),
    }
    componentWillMount() {
        const {
            searchName, dateStart, dateEnd, limit,
        } = this.props;

        this.setState({
            searchName, dateStart, dateEnd, limit,
        });
    }
    submitSearchHotels = (e) => {
        e.preventDefault();
        const { requestListHotels } = this.props;
        const {
            searchName, dateStart, dateEnd, limit,
        } = this.state;

        requestListHotels(searchName, dateStart, dateEnd, limit);
    }
    onChange = (e) => {
        const { target: { name, value } } = e;
        const changeState = { [name]: value };

        if (name === 'dateStart') {
            changeState.minEndDate = helperFunction.changeFormatDate(new Date(value), 1);
        }
        this.setState(changeState);
    }
    render() {
        const {
            searchName, dateStart, dateEnd, minStartDate, minEndDate, limit,
        } = this.state;

        return (
            <form onSubmit={this.submitSearchHotels} className='filterPanel'>
                <Field
                    name='searchName'
                    value={searchName}
                    onChange={this.onChange}
                    className='searchField'
                    required
                />
                <Field
                    type='date'
                    name='dateStart'
                    min={minStartDate}
                    value={dateStart}
                    onChange={this.onChange}
                    className='startDateField'
                />
                <Field
                    type='date'
                    name='dateEnd'
                    min={minEndDate}
                    value={dateEnd}
                    onChange={this.onChange}
                    className='endDateField'
                />
                <Field
                    type='number'
                    name='limit'
                    min='1'
                    max='10'
                    value={limit}
                    onChange={this.onChange}
                />
                <button type='submit' className='searchButton'>search</button>
            </form>
        );
    }
}

FilterPanel.propTypes = {
    requestListHotels: PropTypes.func,
    searchName: PropTypes.string,
    dateStart: PropTypes.string,
    dateEnd: PropTypes.string,
    limit: PropTypes.string,
};

export default FilterPanel;
