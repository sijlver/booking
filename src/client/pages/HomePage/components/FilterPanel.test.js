import React from 'react';
import { shallow  } from "enzyme";
import jest from 'jest-mock';

import FilterPanel from './FilterPanel';

describe('FilterPanel', () => {
    it('should render self', () => {
        const props = {
            searchName: 'searchName',
            dateStart: '10-08-1991',
            dateEnd: '11-12-1991',
            limit: '10',
            requestListHotels: jest.fn()
        };
        const wrapper = shallow (
            <FilterPanel {...props} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
