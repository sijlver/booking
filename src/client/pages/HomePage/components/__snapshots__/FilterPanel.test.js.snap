// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel should render self 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterPanel
    dateEnd="11-12-1991"
    dateStart="10-08-1991"
    limit="10"
    requestListHotels={[MockFunction]}
    searchName="searchName"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Field
          className="searchField"
          name="searchName"
          onChange={[Function]}
          required={true}
          value="searchName"
        />,
        <Field
          className="startDateField"
          min="2018-05-25"
          name="dateStart"
          onChange={[Function]}
          type="date"
          value="10-08-1991"
        />,
        <Field
          className="endDateField"
          min="2018-05-26"
          name="dateEnd"
          onChange={[Function]}
          type="date"
          value="11-12-1991"
        />,
        <Field
          max="10"
          min="1"
          name="limit"
          onChange={[Function]}
          type="number"
          value="10"
        />,
        <button
          className="searchButton"
          type="submit"
        >
          search
        </button>,
      ],
      "className": "filterPanel",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "searchField",
          "name": "searchName",
          "onChange": [Function],
          "required": true,
          "value": "searchName",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "startDateField",
          "min": "2018-05-25",
          "name": "dateStart",
          "onChange": [Function],
          "type": "date",
          "value": "10-08-1991",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "endDateField",
          "min": "2018-05-26",
          "name": "dateEnd",
          "onChange": [Function],
          "type": "date",
          "value": "11-12-1991",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "max": "10",
          "min": "1",
          "name": "limit",
          "onChange": [Function],
          "type": "number",
          "value": "10",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "search",
          "className": "searchButton",
          "type": "submit",
        },
        "ref": null,
        "rendered": "search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Field
            className="searchField"
            name="searchName"
            onChange={[Function]}
            required={true}
            value="searchName"
          />,
          <Field
            className="startDateField"
            min="2018-05-25"
            name="dateStart"
            onChange={[Function]}
            type="date"
            value="10-08-1991"
          />,
          <Field
            className="endDateField"
            min="2018-05-26"
            name="dateEnd"
            onChange={[Function]}
            type="date"
            value="11-12-1991"
          />,
          <Field
            max="10"
            min="1"
            name="limit"
            onChange={[Function]}
            type="number"
            value="10"
          />,
          <button
            className="searchButton"
            type="submit"
          >
            search
          </button>,
        ],
        "className": "filterPanel",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "searchField",
            "name": "searchName",
            "onChange": [Function],
            "required": true,
            "value": "searchName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "startDateField",
            "min": "2018-05-25",
            "name": "dateStart",
            "onChange": [Function],
            "type": "date",
            "value": "10-08-1991",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "endDateField",
            "min": "2018-05-26",
            "name": "dateEnd",
            "onChange": [Function],
            "type": "date",
            "value": "11-12-1991",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "max": "10",
            "min": "1",
            "name": "limit",
            "onChange": [Function],
            "type": "number",
            "value": "10",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "search",
            "className": "searchButton",
            "type": "submit",
          },
          "ref": null,
          "rendered": "search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
